include/rig.hrl

```diff
@@ -23,7 +23,7 @@
 %% types
 -type basedir() :: string().
 -type config()  :: {table(), file(), decoder(), options()}.
--type decoder() :: fun((binary()) -> tuple()) | term.
+-type decoder() :: fun((binary()) -> tuple()) | term | {module, module()}.
```

Comment:
Out of curiosity, why only provide the module name and not allow for `{module(), function()}` instead. Seems like the later gives more flexibility to the user on how they want to organize their code while not having any downside performance-wise.

Reply:
I tend to agree that this is a better solution.

Reply:
fixed